integration/feature/init/src/MillInitSbtTests.scala

```diff
@@ -0,0 +1,230 @@
+
+      val compileResult = eval("compile")
+      assert(compileResult.isSuccess)
+    }
+  }
+}
+
+object MillInitScalaCsv200Tests extends BuildGenTestSuite {
+  def tests: Tests = Tests {
+    /*
+    - 34 KB
+    - originally sbt 1.10.0
+     */
+    val url = "https://github.com/tototoshi/scala-csv/archive/refs/tags/2.0.0.zip"
+
+    test - integrationTest(url) { tester =>
+      import tester.*
+
+      bumpSbtTo1107(workspacePath)
+
+      val initResult = eval(initCommand, stdout = os.Inherit, stderr = os.Inherit)
+      assert(initResult.isSuccess)
+
+      val compileResult = eval("compile")
+      // Cross builds are not supported yet.
+      assert(!compileResult.isSuccess)
+    }
+  }
+}
+
+object MillInitScalaCsv136Tests extends BuildGenTestSuite {
+  def tests: Tests = Tests {
+    /*
+    - 28 KB
+    - originally sbt 1.2.8
+     */
+    val url = "https://github.com/tototoshi/scala-csv/archive/refs/tags/1.3.6.zip"
+
+    test - integrationTest(url) { tester =>
+      import tester.*
+
+      bumpSbtTo1107(workspacePath)
+
+      val initResult = eval(initCommand, stdout = os.Inherit, stderr = os.Inherit)
+      assert(initResult.isSuccess)
+
+      val compileResult = eval("compile")
+      // It works here, but scala 2.11 with JDK 6 seems not supported when run with JDK 17 in shell.
+      assert(compileResult.isSuccess)
+    }
+  }
+}
+
+// same as the one in the unit tests
+object MillInitSbtMultiProjectExampleTests extends BuildGenTestSuite {
+  def tests: Tests = Tests {
+    /*
+    - 12 KB
+    - originally sbt 1.0.2
+     */
+    val url =
+      "https://github.com/pbassiner/sbt-multi-project-example/archive/152b31df9837115b183576b0080628b43c505389.zip"
+
+    test - integrationTest(url) { tester =>
+      import tester.*
+
+      bumpSbtTo1107(workspacePath)
+
+      val initResult = eval(initCommand, stdout = os.Inherit, stderr = os.Inherit)
+      assert(initResult.isSuccess)
+
+      val compileResult = eval("compile")
+      assert(compileResult.isSuccess)
+
+      val compileSubmodulesResult = eval("_.compile")
+      if (System.getProperty("java.version").split('.').head.toInt <= 11)
+        assert(compileSubmodulesResult.isSuccess)
+      else
+        // Submodules don't compile well with JDK 17 and 21, which seems to be due to incompatible bytecode versions in dependencies.
+        assert(!compileSubmodulesResult.isSuccess)
+    }
+  }
+}
+
+// relatively large libraries
+
+object MillInitZioHttpTests extends BuildGenTestSuite {
+  def tests: Tests = Tests {
+    /*
+    - 1.4 MB
+    - originally sbt 1.10.0
+     */
+    val url = "https://github.com/zio/zio-http/archive/refs/tags/v3.0.1.zip"
+
+    test - integrationTest(url) { tester =>
+      import tester.*
+
+      bumpSbtTo1107(workspacePath)
+
+      val initResult = eval(initCommand, stdout = os.Inherit, stderr = os.Inherit)
+      assert(initResult.isSuccess)
+
+      val compileResult = eval("compile")
+      assert(compileResult.isSuccess)
```

Comment:
Apart from `compile` on the root module, does `.compile` work on any other modules? which ones?

Reply:
I will resolve these after fixing https://github.com/com-lihaoyi/mill/pull/4586#discussion_r1965045922.

Reply:
As expained in https://github.com/com-lihaoyi/mill/pull/4586#discussion_r1970355857.